packages/@ember/-internals/glimmer/lib/components/link-to.ts

```diff
@@ -418,40 +418,6 @@ if (EMBER_MODERNIZED_BUILT_IN_COMPONENTS) {
     });
   }
 
-  // @tagName
-  {
-    let superOnUnsupportedArgument = prototype['onUnsupportedArgument'];
-
-    Object.defineProperty(prototype, 'onUnsupportedArgument', {
-      configurable: true,
-      enumerable: false,
-      value: function onUnsupportedArgument(this: LinkTo, name: string): void {
-        if (name === 'tagName') {
-          let tagName = this.named('tagName');
-
-          deprecate(
-            `Passing the \`@tagName\` argument to <LinkTo> is deprecated. Using a <${tagName}> ` +
-              'element for navigation is not recommended as it creates issues with assistive ' +
-              'technologies. Remove this argument to use the default <a> element. In the rare ' +
-              'cases that calls for using a different element, refactor to use the router ' +
-              'service inside a custom event handler instead.',
-            false,
-            {
-              id: 'ember.link-to.tag-name',
-              for: 'ember-source',
-              since: {},
-              until: '4.0.0',
-            }
-          );
-
-          this.modernized = false;
```

Comment:
It is asserted that passing `<LinkTo @tagName="div">` still renders an `<a>` after this removal?

Reply:
[This bit of code](https://github.com/emberjs/ember.js/blob/master/packages/@ember/-internals/glimmer/lib/components/internal.ts#L256) causes any test I write to fail due to `<unsupportedProp> is deprecated` so I think it's pretty well aserted

Reply:
Good call. I added an `assert` to the `validateArguments` method and some tests that assert for this error, but not sure if that's the right place to throw versus where [`@href` throws](https://github.com/emberjs/ember.js/blob/master/packages/@ember/-internals/glimmer/lib/components/link-to.ts#L404-L419)